rust/src/metadata.rs

```diff
@@ -219,6 +219,32 @@ impl TransactionMetadatum {
     }
 }
 
+impl serde::Serialize for TransactionMetadatum {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: serde::Serializer {
+        let json_str = decode_metadatum_to_json_str(self, MetadataJsonSchema::DetailedSchema)
+            .map_err(|e| serde::ser::Error::custom(&format!("{:?}", e)))?;
+        serializer.serialize_str(&json_str)
+    }
+}
+
+impl <'de> serde::de::Deserialize<'de> for TransactionMetadatum {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where
+    D: serde::de::Deserializer<'de> {
+        let s = <String as serde::de::Deserialize>::deserialize(deserializer)?;
+        encode_json_str_to_metadatum(s.clone(), MetadataJsonSchema::DetailedSchema)
+            .map_err(|e| serde::de::Error::invalid_value(serde::de::Unexpected::Str(&s), &format!("{:?}", e).as_str()))
+    }
+}
+
+// just for now we'll do json-in-json until I can figure this out better
+// TODO: maybe not generate this? or how do we do this?
+impl JsonSchema for TransactionMetadatum {
```

Comment:
This is not the best solution but the options are either:
1) Just make our own (incompatible with cardano-node) JSON rep using schemars/serde
2) Manually write .ts defs for the cardano-node JSON schema we follow (I picked `DetailedSchema` since it supports everything (besides edge-cases with `i64` in serde-json) and merge those in as a final step. We'd want to do the same with plutus datums I guess if we go this route.

I'm leaning towards 2 but wanted to get opinions on it. We might have to write them as JSON schemas and generate from there and hope we can get it to match the .ts we want as otherwise other schemas referring to metadata (e.g. tx) might break.